pagopa/io-app
# chore: [IOBP-466] Add gallery permission request in barcode scan screen for iOS devices

⚠️ Depends on https://github.com/pagopa/io-app/pull/5313

## Short description
This PR adds the image gallery permission request in barcode scan screen for iOS devices

## List of changes proposed in this pull request
- Added `react-native-permissions` package
- Added `react-native-permissions` configuration in `Podfile`
- Updated `@pagopa/io-app-design-system` to v 1.19.0
- Removed `react-native` and `@types/react-native` patches for Android media permissions, now being managed by `react-native-permissions` library
- Replaced permission request functions in `ts/utils/permissions.ts` with new implementation using `react-native-permissions`
- Renamed `AndroidMediaPermissionRequestScreen ` to `GalleryPermissionInstructionsScreen`, which now displays instructions for both Android and iOS devices

## How to test
Check the following with a fresh install on both iOS and Android devices:
- navigate to the barcode scan screen and tap on **Upload > Upload image**, a prompt should be shown to request gallery permissions. If permissions are denied, a screen with instructions is shown
- navigate to the **Profile > Preferences** section, tap on **Preferred Calendar**, a prompt should be shown to request calendar permissions

## Preview

https://github.com/pagopa/io-app/assets/6160324/2ec97018-261a-4323-91d8-15a194841773
